File: docs/legacy(old PACE)/C#/PaceAPISample/sample/GeoLocateContacts.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using efipaceservices;

namespace Pace_Web_Service_SDK.sample
{
    class GeoLocateContacts
    {
        public static void Run()
        {
            String latitude = "40.77447";
            String longitude = "-73.96150";
            int radius = 50;
            String xpath = "@active = 'true'";

            new GeoLocateContacts().run(latitude, longitude, radius, xpath);
        }

        public void run(String latitude, String longitude, int radius, String xpathExpression)
        {
            GeoLocatePortType geoLocateHttpBinding = PaceClient.getGeoLocateHttpBinding();
            ArrayOfString3 keys = geoLocateHttpBinding.findContacts(latitude, longitude, radius, xpathExpression);

            Console.WriteLine("Following contacts in range: ");

            foreach (String contact in keys)
            {
                Console.WriteLine(contact);
            }
        }

    }
}

File: docs/legacy(old PACE)/unsupported/VB.NET/PaceWebServiceSDK/sample/GeoLocateContacts.vb
Imports PaceWebServiceSDK.efipaceservices

Public Class GeoLocateContacts
    Public Shared Sub Run()
        Dim latitude As String = "30"
        Dim longitude As String = "-81.5"

        Dim radius As Integer = 50
        Dim xpath As String = "@firstName = 'EFI Pace'"

        findContacts(latitude, longitude, radius, xpath)
    End Sub

    Private Shared Sub findContacts(ByVal latitude As String, ByVal longitude As String, ByVal radius As Integer, ByVal xpathExpression As String)

        Dim geoLocateHttpBinding As GeoLocateHttpBinding = PaceClient.getGeoLocateHttpBinding()
        Dim keys As [String]() = geoLocateHttpBinding.findContacts(latitude, longitude, radius, xpathExpression)


        Console.WriteLine("Following contacts in range: ")

        For Each contact As [String] In keys
            Console.WriteLine(contact)
        Next
    End Sub

End Class
